drivers/serial/uart_ns16550.c

```diff
@@ -622,6 +622,11 @@ static int uart_ns16550_configure(const struct device *dev,
 			goto out;
 		}
 
+		ret = clock_control_on(dev_cfg->clock_dev, dev_cfg->clock_subsys);
+		if (ret != 0 && ret != -EALREADY) {
+			goto out;
+		}
```

Comment:
Don't you need to reset `err` to `0` in the `-EALREADY` case? Otherwise the function may continue successfully, but ultimately return `-EALREADY` at the very end.

Reply:
I think the perhaps cleaner way would be to remove the pre-initialization of `ret` from line 578, and instead do `ret = 0;` right before the `out:` label. Right now all the places that do `goto out;` explicitly set `ret` first.